rskj-core/src/main/java/co/rsk/peg/bitcoin/FlyoverRedeemScriptBuilderImpl.java

```diff
@@ -40,4 +44,26 @@ private void validateFlyoverDerivationHash(Keccak256 flyoverDerivationHash) {
             throw new RedeemScriptCreationException(message, INVALID_FLYOVER_DERIVATION_HASH);
         }
     }
+
+    private void validateRedeemScript(Script redeemScript) {
+        if (isNull(redeemScript)) {
+            String message = "Provided redeem script is null.";
+            logger.warn("[validateRedeemScript] {}", message);
+            throw new RedeemScriptCreationException(message, INVALID_INTERNAL_REDEEM_SCRIPTS);
+        }
+
+        try {
+            RedeemScriptParser redeemScriptParser = RedeemScriptParserFactory.get(
+                redeemScript.getChunks());
+            if (redeemScriptParser.getMultiSigType() == RedeemScriptParser.MultiSigType.FLYOVER) {
+                String message = "Provided redeem script cannot be a flyover redeem script.";
+                logger.warn("[validateRedeemScript] {}", message);
+                throw new RedeemScriptCreationException(message, INVALID_INTERNAL_REDEEM_SCRIPTS);
+            }
+        } catch (Exception e) {
```

Comment:
Should we catch Exception or ScriptException?

Reply:
I think we should catch`Exception` here because any problem while parsing the redeem script basically means we are dealing with an `INVALID_INTERNAL_REDEEM_SCRIPTS`.